.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,94 @@
+name: Build Docker Image
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
+        uses: sigstore/cosign-installer@e1523de7571e31dbe865fd2e80c5c7c23ae71eb4 #v3.4.0
+        with:
+          cosign-release: 'v2.2.3'
+
+      # Set up BuildKit Docker container builder to be able to build
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Extract metadata (tags, labels) for Docker
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      # Build and push Docker image with Buildx (don't push on PR)
+      # https://github.com/docker/build-push-action
+      - name: Build and push Docker image
+        id: build-and-push
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+        with:
+          context: .
```

Comment:
Building for multiple platforms is nice to have, as it allows easier usage on more servers. (Also lets people run it on a raspberry pi, etc.)
```suggestion
          platforms: linux/amd64,linux/arm64
          context: .
```

Reply:
gimme a minute to mess around with this

Reply:
@Zaprit & @sudokoko  it gives an exception during the "restore" phase, i don't know that much dotnet so im not sure exactly whats going on
![Screenshot 2024-03-29 at 10 03 58 AM](https://github.com/LBPUnion/ProjectLighthouse/assets/96893300/7b08d41b-9e74-4647-b6ea-d8868f1e590e)

Reply:
logs: https://github.com/ren-makes-things/ProjectLighthouse/actions/runs/8483485830/job/23244693081

Reply:
im going to rerun the job, see if it was just a fluke

Reply:
fail's again, not sure whats happening, https://github.com/ren-makes-things/ProjectLighthouse/actions/runs/8483485830/job/23244818665

Reply:
just looking at other pipelines that build for arm, seems like you need to do a setup qemu action, just before the buildx setup
```
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v3
```

Reply:
i'll give that a try and report back

Reply:
https://github.com/ren-makes-things/ProjectLighthouse/actions/runs/8483690691/job/23245230894

Reply:
@Zaprit any updates or anything i need to change on my end?

Reply:
hmm, not sure at this point, seems to be a bug in .NET, there's no easy fix it seems, if you just remove the arm thing for now, I'll revisit it later.

Reply:
i'll remove it for now

Reply:
done